lib/routes/emails.js

```diff
@@ -429,7 +430,9 @@ module.exports = (log, db, mailer, config, customs, push) => {
                         // Force it so that we emit the appropriate newsletter state.
                         marketingOptIn: marketingOptIn || false,
                         uid
-                      })
+                      }),
+                      reminder ? request.emitMetricsEvent('account.reminder', { uid }) : null,
+                      verificationReminders.delete(uid),
```

Comment:
This would attempt to delete a reminder even if the user did not have one. Are there any performance implications of this?

Reply:
Yeah, it's O(N), but I haven't actually done the experiment to see what that means in terms of milliseconds with our upper limit of signups in a 5-day period. I'll crunch the numbers tomorrow morning.

Reply:
(and if it does turn out to be too slow, it's straightforward enough to just maintain an inverted index under a different key that we can use in combination with `zremrangebyrank`)

Reply:
> O(N)

Actually, only just noticed the Redis docs include the time complexity, which is nice. `zrem` is O(log N), not O(N), so I think we're going to be good here (but I'll still test).

Reply:
Just to follow up on this thread, I ran some tests with [this script](https://gist.github.com/philbooth/60cea9c1fce35a2cb18f4681ff08fa48):

```js
'use strict';

process.env.NODE_ENV = 'dev';

const config = require('./config').getProperties();
const crypto = require('crypto');
const log = require('./test/mocks').mockLog();
const redis = require('fxa-shared/redis')({
  ...config.redis,
  ...config.verificationReminders.redis,
  enabled: true,
  prefix: 'wibble:',
}, log);

const KEY = 'test';

test();

async function test () {
  await redis.zpoprangebyscore(KEY, 0, Date.now());

  let first, last;
  for (let i = 0; i < 1000000; ++i) {
    const item = crypto.randomBytes(64).toString('hex');
    switch (i) {
      case 0:
        first = item;
        break;
      case 999999:
        last = item;
    }
    await redis.zadd(KEY, Date.now(), item);
  }

  let before, result, after;

  before = Date.now();
  result = await redis.zrem(KEY, 'blee');
  after = Date.now();
  console.log('zrem non-existent:', after - before, result === 0);

  before = Date.now();
  result = await redis.zrem(KEY, first);
  after = Date.now();
  console.log('zrem first:', after - before, result === 1);

  before = Date.now();
  result = await redis.zrem(KEY, last);
  after = Date.now();
  console.log('zrem last:', after - before, result === 1);
}
```

I expect our peak verification reminder count to be less than 100,000 items but to err on the side of caution, I tested against 1,000,000 items. `zrem` posts sub-millisecond timings in every case on my machine.

Reply:
> I expect our peak verification reminder count to be less than 100,000 items...

Correction, I was looking at the wrong metric. I expect it to be less than 500,000 items now.